xattree/__init__.py

```diff
@@ -1264,7 +1264,7 @@ def _transform_field(field: Attribute) -> Attribute:
                         pass
 
                 if not (
-                    attrs_has(type_)
+                    has_xats(type_)
```

Comment:
this was the fix for the main issue, the change above are for the second issue described